web/apps/labelstudio/src/pages/CreateProject/Config/Preview.jsx

```diff
@@ -9,15 +9,21 @@ import { useAPI } from "../../../providers/ApiProvider";
 
 const configClass = cn("configure");
 
-const loadDependencies = async () => import("@humansignal/editor");
+let dependencies;
+const loadDependencies = async () => {
+  if (!dependencies) {
+    dependencies = import("@humansignal/editor");
+  }
+  return dependencies;
+};
 
 export const Preview = ({ config, data, error, loading, project }) => {
+  loadDependencies();
```

Comment:
this way of calling it doesn't feel correct... besides of the comment from ellipsis it's also called synchronously.
or I'm just lacking some knowledge of dynamic `import()`. if you call it repeatedly and in parallel would the module be imported only once and the same instance will be returned in different places?

Reply:
>if you call it repeatedly and in parallel would the module be imported only once and the same instance will be returned in different places?

Technically there is no parallel capability but the dynamic import will happen only once, right on the first render cycle of the component. That starts the async load, which then is awaited in the useEffect to ensure it completes. It all references the same promise, and only assigns it once.

This is basically the loader pattern introduced with React 18, and Suspense, but without all that jazz.

Reply:
As mentioned in DM, we can go with a different approach if you are feeling this is not resonating.

Reply:
As discussed, added a comment here about what this is doing.